### ISSUE TITLE
Invalid Argument: Invalid numeric value after Typed Link migration

### ISSUE THREAD

**Issue by nixondesigndev** [verbb/hyper#10]
### Describe the bug

After migrating from Typed Link field and visitng an elements edit page an Invalid arguement exception is thrown. The error is coming from a Hyper/Typed Link field within a SuperTable field within a matrix.

### Steps to reproduce

1. Create a Typed Link field inside a SuperTable field, inside a Matrix field
2. Assign field to an Element's field layout
3. Populate Typed Link field
4. Run Hyper migration
5. Visit elements edit page

### Craft CMS version

4.3.9

### Plugin version

1.0.2

### Multi-site?

No

### Additional context

Stack track
```
yii\base\InvalidArgumentException: Invalid numeric value: https://craft.ddev.site/work/ in /var/www/html/vendor/craftcms/cms/src/helpers/Db.php:578
Stack trace:
#0 /var/www/html/vendor/craftcms/cms/src/helpers/Db.php(825): craft\helpers\Db::parseParam('elements.id', Array, '=', false, 'integer')
#1 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1330): craft\helpers\Db::parseNumericParam('elements.id', 'https://craft.d...')
#2 /var/www/html/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\mysql\QueryBuilder))
#3 /var/www/html/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build(Object(craft\elements\db\EntryQuery))
#4 /var/www/html/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand(Object(craft\db\Connection))
#5 /var/www/html/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one(NULL)
#6 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1506): craft\db\Query->one(NULL)
#7 /var/www/html/vendor/verbb/hyper/src/base/ElementLink.php(161): craft\elements\db\ElementQuery->one()
#8 /var/www/html/vendor/verbb/hyper/src/base/ElementLink.php(175): verbb\hyper\base\ElementLink->getElement()
#9 /var/www/html/vendor/verbb/hyper/src/base/Link.php(355): verbb\hyper\base\ElementLink->getLinkUrl()
#10 /var/www/html/vendor/verbb/hyper/src/base/Link.php(118): verbb\hyper\base\Link->getUrl()
#11 /var/www/html/vendor/verbb/hyper/src/base/Link.php(149): verbb\hyper\base\Link->__toString()
#12 /var/www/html/vendor/verbb/hyper/src/base/Link.php(154): verbb\hyper\base\Link->count()
#13 /var/www/html/vendor/verbb/hyper/src/models/LinkCollection.php(121): verbb\hyper\base\Link->isEmpty()
#14 /var/www/html/vendor/verbb/hyper/src/fields/HyperField.php(433): verbb\hyper\models\LinkCollection->isEmpty()
#15 /var/www/html/vendor/verbb/hyper/src/fields/HyperField.php(172): verbb\hyper\fields\HyperField->_getLinksForInput(Object(verbb\hyper\models\LinkCollection))
#16 /var/www/html/vendor/twig/twig/src/Extension/CoreExtension.php(1607): verbb\hyper\fields\HyperField->getInputHtml(Object(verbb\hyper\models\LinkCollection), Object(verbb\supertable\elements\SuperTableBlockElement))
#17 /var/www/html/vendor/craftcms/cms/src/helpers/Template.php(110): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(verbb\hyper\fields\HyperField), 'getInputHtml', Array, 'method', false, false)
#18 /var/www/html/storage/runtime/compiled_templates/89/894f10063c2b33c4247987a67e848524.php(117): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(verbb\hyper\fields\HyperField), 'getInputHtml', Array, 'method')
#19 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_956253f2acfc81dcb176078391e4fb77->doDisplay(Array, Array)
#20 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#21 /var/www/html/storage/runtime/compiled_templates/e5/e5c6a81a044f1a0965aa9079bdd06cd5.php(139): Twig\Template->display(Array)
#22 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_5b833dbe758761de081acd6a018cbebb->doDisplay(Array, Array)
#23 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#24 /var/www/html/storage/runtime/compiled_templates/64/649744c25fbd74d0b4a982eb73c66ded.php(125): Twig\Template->display(Array)
#25 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_a199846ca781416a0c3a06185b663a8e->doDisplay(Array, Array)
#26 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#27 /var/www/html/storage/runtime/compiled_templates/55/55c4c6e72d5aeee4f783d9664f25a949.php(43): Twig\Template->display(Array)
#28 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_4607e995bb40761b9c34a2562b3c7d2d->doDisplay(Array, Array)
#29 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#30 /var/www/html/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#31 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#32 /var/www/html/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#33 /var/www/html/vendor/craftcms/cms/src/web/View.php(451): Twig\Environment->render('super-table/inp...', Array)
#34 /var/www/html/vendor/verbb/super-table/src/fields/SuperTableField.php(778): craft\web\View->renderTemplate('super-table/inp...', Array)
#35 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(269): verbb\supertable\fields\SuperTableField->getInputHtml(Array, Object(craft\elements\MatrixBlock))
#36 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\MatrixBlock), false)
#37 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(214): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\MatrixBlock), false)
#38 /var/www/html/vendor/craftcms/cms/src/web/View.php(1611): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#39 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(213): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#40 /var/www/html/vendor/craftcms/cms/src/models/FieldLayout.php(739): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\MatrixBlock), false)
#41 /var/www/html/vendor/craftcms/cms/src/web/View.php(1605): craft\models\FieldLayout->craft\models\{closure}()
#42 /var/www/html/vendor/craftcms/cms/src/models/FieldLayout.php(738): craft\web\View->namespaceInputs(Object(Closure), NULL)
#43 /var/www/html/vendor/twig/twig/src/Extension/CoreExtension.php(1607): craft\models\FieldLayout->createForm(Object(craft\elements\MatrixBlock), false)
#44 /var/www/html/vendor/craftcms/cms/src/helpers/Template.php(110): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method', false, false)
#45 /var/www/html/storage/runtime/compiled_templates/3a/3ace5d39e120d44fc4cc8b202f727573.php(293): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\models\FieldLayout), 'createForm', Array, 'method')
#46 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_c3268286598198c454ffd7fb0bb80c98->doDisplay(Array, Array)
#47 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#48 /var/www/html/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#49 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#50 /var/www/html/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#51 /var/www/html/vendor/craftcms/cms/src/web/View.php(451): Twig\Environment->render('_components/fie...', Array)
#52 /var/www/html/vendor/craftcms/cms/src/fields/Matrix.php(739): craft\web\View->renderTemplate('_components/fie...', Array)
#53 /var/www/html/vendor/craftcms/cms/src/base/Field.php(495): craft\fields\Matrix->inputHtml(Array, Object(craft\elements\Entry))
#54 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(269): craft\base\Field->getInputHtml(Object(craft\elements\db\MatrixBlockQuery), Object(craft\elements\Entry))
#55 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/BaseField.php(215): craft\fieldlayoutelements\CustomField->inputHtml(Object(craft\elements\Entry), false)
#56 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(214): craft\fieldlayoutelements\BaseField->formHtml(Object(craft\elements\Entry), false)
#57 /var/www/html/vendor/craftcms/cms/src/web/View.php(1611): craft\fieldlayoutelements\CustomField->craft\fieldlayoutelements\{closure}()
#58 /var/www/html/vendor/craftcms/cms/src/fieldlayoutelements/CustomField.php(213): craft\web\View->namespaceInputs(Object(Closure), 'fields')
#59 /var/www/html/vendor/craftcms/cms/src/models/FieldLayout.php(739): craft\fieldlayoutelements\CustomField->formHtml(Object(craft\elements\Entry), false)
#60 /var/www/html/vendor/craftcms/cms/src/web/View.php(1605): craft\models\FieldLayout->craft\models\{closure}()
#61 /var/www/html/vendor/craftcms/cms/src/models/FieldLayout.php(738): craft\web\View->namespaceInputs(Object(Closure), NULL)
#62 /var/www/html/vendor/craftcms/cms/src/controllers/ElementsController.php(744): craft\models\FieldLayout->createForm(Object(craft\elements\Entry), false, Array)
#63 /var/www/html/vendor/craftcms/cms/src/controllers/ElementsController.php(414): craft\controllers\ElementsController->_prepareEditor(Object(craft\elements\Entry), true, Object(craft\web\Response), 'main-form', Object(Closure), Object(Closure), Object(Closure))
#64 [internal function]: craft\controllers\ElementsController->craft\controllers\{closure}(Object(craft\web\Response), 'main-form')
#65 /var/www/html/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(113): call_user_func(Object(Closure), Object(craft\web\Response), 'main-form')
#66 /var/www/html/vendor/craftcms/cms/src/web/CpScreenResponseFormatter.php(48): craft\web\CpScreenResponseFormatter->_formatTemplate(Object(craft\web\Response), Object(craft\web\CpScreenResponseBehavior))
#67 /var/www/html/vendor/yiisoft/yii2/web/Response.php(1098): craft\web\CpScreenResponseFormatter->format(Object(craft\web\Response))
#68 /var/www/html/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#69 /var/www/html/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#70 /var/www/html/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#71 /var/www/html/public/index.php(12): yii\base\Application->run()
#72 {main}
```

**Comment by engram-design** [comment]
I can't seem to immediately replicate this I'm afraid. I've tried several combinations of Super Table/Matrix fields as well.

Where there any errors during the migration?

For some reason, the value of the field is `https://craft.ddev.site/work/` when it should be the ID of the element.

Any chance you can send through your `lenz_linkfield` db table to `support@verbb.io`?

I would also roll back your database to the backup taken before the migration if you continue to want to develop the site. We cannot run the migration again effectively if it's already processed fields and content.

**Comment by engram-design** [comment]
Finally able to get to the bottom of this with your help! Turns out, a really specific set of circumstances with Matrix fields, and having multiple link fields in a single Matrix field (across different blocktypes).

Should be fixed in [1.0.3](https://github.com/verbb/hyper/releases/tag/1.0.3).

As I mentioned, please roll back your database to before you ran the Hyper migration (so your fields are still on Typed Link), and run the migration again.

**Comment by jan-dh** [comment]
@engram-design I'm still running into template rendering erros: `Invalid numeric values` on the latest version of Hyper.

**Comment by engram-design** [comment]
Something must have gone wrong with a migration - presumably from Typed Link like the OP? I'm also added handling to discord a non-numeric value for an element-based link, to get around this error. To get this early run `composer require verbb/hyper:"dev-craft-4 as 1.1.16"`

**Comment by jan-dh** [comment]
The fields weren't migrated, hyper from the start.

**Comment by engram-design** [comment]
Yep, regardless that should fix things

**Comment by jan-dh** [comment]
@engram-design I've pushed this update to production and it removed all existing entry links 😳

**Comment by engram-design** [comment]
Sorry about that, just pushed a fix for that

**Comment by engram-design** [comment]
Fixed in [1.1.17](https://github.com/verbb/hyper/releases/tag/1.1.17)
